q2cli/builtin/tools.py

```diff
@@ -450,20 +450,10 @@ def view(visualization_path, index_extension):
             'https://view.qiime2.org, or move the Visualization to an '
             'environment with a display and view it with `qiime tools view`.')
 
-    import zipfile
-    import qiime2.sdk
-
     if index_extension.startswith('.'):
         index_extension = index_extension[1:]
-    try:
-        visualization = qiime2.sdk.Visualization.load(visualization_path)
-    # TODO: currently a KeyError is raised if a zipped file that is not a
-    # QIIME 2 result is passed. This should be handled better by the framework.
-    except (zipfile.BadZipFile, KeyError, TypeError):
-        raise click.BadParameter(
-            '%s is not a QIIME 2 Visualization. Only QIIME 2 Visualizations '
-            'can be viewed.' % visualization_path)
 
+    visualization = _load_input(visualization_path, view=True)[0]
```

Comment:
I think we still need to assert that `isinstance(visualization, qiime2.Visualization)` is true, and otherwise raise the original error.

Reply:
Oh yeah good catch